### PR TITLE
Nrunner support to timeout class attribute

### PR DESCRIPTION
**Pull request by richtja**
It adds support to the timeout class attribute for avocado instrumented tests in nrunner.

Reference:#4257
Signed-off-by: Jan Richter <jarichte@redhat.com>

### CODE REVIEW

File: avocado/core/nrunner_avocado_instrumented.py [hunk 1]
```diff
@@ -70,9 +73,17 @@ def run(self):
...
+        time_started = time.monotonic()
         yield self.prepare_status('started')
+
+        # Waiting for early status
+        while queue.empty():
+            time.sleep(nrunner.RUNNER_RUN_CHECK_INTERVAL)
         most_current_execution_state_time = None
+        early_status = queue.get()
+        timeout = early_status['timeout']
+        timeout = float(timeout or self.DEFAULT_TIMEOUT)
```
Comment (beraldoleal) [inline avocado/core/nrunner_avocado_instrumented.py:85]:
```python 
timeout = float(early_status.get('timeout', self.DEFAULT_TIMEOUT))
```

**Review by beraldoleal — CHANGES_REQUESTED** [review]

**Comment by richtja** [comment]
@beraldoleal Thanks for your comment, I will push the change here.

File: avocado/core/nrunner_avocado_instrumented.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ class and method names should be separated by a ":".  One
 
      * args: not used
     """
+    DEFAULT_TIMEOUT = 86400
```
Comment (beraldoleal) [inline avocado/core/nrunner_avocado_instrumented.py:23]: How could we make this a custom setting by the users? This is something that I'm trying to do for the "skip exit codes".

**Review by beraldoleal — APPROVED** [review]

**Review by willianrampazzo — APPROVED** [review]
Tested and it works as expected.

**Pull request merged**